Index: kernel/classes/ezorderitem.php
===================================================================
--- kernel/classes/ezorderitem.php	(Revision 22135)
+++ kernel/classes/ezorderitem.php	(Arbeitskopie)
@@ -73,7 +73,7 @@
                                                            'default' => 0,
                                                            'required' => true ),
                                          'vat_value' => array( 'name' => 'VATValue',
-                                                               'datatype' => 'integer',
+                                                               'datatype' => 'float',
                                                                'default' => 0,
                                                                'required' => true ),
                                          'is_vat_inc' => array( 'name' => 'IsVATIncluded',
