Index: kernel/classes/ezorder.php
===================================================================
--- kernel/classes/ezorder.php	(revision 21329)
+++ kernel/classes/ezorder.php	(working copy)
@@ -309,33 +309,43 @@
      \static
      \return the number of active orders
     */
-    function &orderStatistics( $year = false, $month = false )
+    function &orderStatistics( $year = false, $startMonth = false, $stopMonth = false, $startDay = false, $stopDay = false )
     {
         if ( $year == false )
         {
             $startDate = 0;
             $stopDate = mktime( 0, 0, 0, 12, 31, 2037 );
         }
-        else if ( $year != false and $month == false )
+        else if ( $year != false and $startMonth == false and $startDay == false and $stopDay == false )
         {
             $nextYear = $year + 1;
             $startDate = mktime( 0, 0, 0, 1, 1, $year );
             $stopDate = mktime( 0, 0, 0, 1, 1, $nextYear );
         }
-        else if ( $year != false and $month != false )
+        else if ( $year != false and $startMonth != false and $startDay == false and $stopDay == false )
         {
-            $nextMonth = $month + 1;
-            $startDate = mktime( 0, 0, 0, $month, 1, $year );
+            $nextMonth = $startMonth + 1;
+            $startDate = mktime( 0, 0, 0, $startMonth, 1, $year );
             $stopDate = mktime( 23, 59, 59, $nextMonth, 0, $year );
         }
+        else if ( $year != false and $startMonth != false and $startDay != false and $stopDay != false )
+        {
+            $startDate = mktime( 0, 0, 0, $startMonth, $startDay, $year );
+            $stopDate = mktime( 23, 59, 59, $stopMonth, $stopDay, $year );
+        }
 
         $db =& eZDB::instance();
-        $productArray = $db->arrayQuery(  "SELECT ezproductcollection_item.*,  ignore_vat, ezorder.created, currency_code FROM ezorder, ezproductcollection_item, ezproductcollection
+        $productArray = $db->arrayQuery(  "SELECT ezproductcollection_item.*, ignore_vat, ezorder.created, ezorder_item.price as order_shipping_price, currency_code FROM ezorder, ezorder_item, ezproductcollection_item, ezproductcollection
                                                 WHERE ezproductcollection.id=ezproductcollection_item.productcollection_id
+                                                  AND ezorder_item.order_id = ezorder.id
                                                   AND ezproductcollection_item.productcollection_id=ezorder.productcollection_id
                                                   AND is_temporary='0'
                                                   AND ezorder.created >= '$startDate' AND ezorder.created < '$stopDate'
                                              ORDER BY contentobject_id, currency_code" );
+
+        $shippingSumSQL = "SELECT sum(ezorder_item.price) as order_shipping_price FROM ezorder, ezorder_item WHERE ezorder_item.order_id = ezorder.id AND ezorder.is_temporary = 0 AND ezorder.created >= '$startDate' AND ezorder.created < '$stopDate'";
+        $shippingSum = $db->arrayQuery( $shippingSumSQL );
+
         $currentContentObjectID = 0;
         $productItemArray = array();
         $statisticArray = array();
@@ -380,12 +390,14 @@
             {
                 $productInfo[$currencyCode] = array( 'sum_count' => 0,
                                                      'sum_ex_vat' => 0,
-                                                     'sum_inc_vat' => 0 );
+                                                     'sum_inc_vat' => 0,
+                                                     'sum_shipping' => 0 );
             }
             if ( !isset( $totalSumInfo[$currencyCode] ) )
             {
                 $totalSumInfo[$currencyCode] = array( 'sum_ex_vat' => 0,
-                                                      'sum_inc_vat' => 0 );
+                                                      'sum_inc_vat' => 0,
+                                                      'sum_shipping' => 0 );
             }
 
             if ( !isset( $totalSumIncVAT[$currencyCode] ) )
@@ -394,6 +406,9 @@
             if ( !isset( $totalSumExVAT[$currencyCode] ) )
                 $totalSumExVAT[$currencyCode] = 0;
 
+            if ( !isset( $totalSumShipping[$currencyCode] ) )
+                $totalSumShipping[$currencyCode] = 0;
+
             if ( $productItem['ignore_vat']== true )
             {
                 $vatValue = 0;
@@ -436,6 +451,10 @@
             $productInfo[$currencyCode]['sum_ex_vat'] += $totalPriceExVAT;
             $productInfo[$currencyCode]['sum_inc_vat'] += $totalPriceIncVAT;
         }
+        
+        $shippingSumNumeric = $shippingSum;
+        $shippingSumNumeric = $shippingSumNumeric[0]['order_shipping_price'];
+        $totalSumInfo[$currencyCode]['sum_shipping'] = $shippingSumNumeric;
 
         // add last product info
         if ( $productCount != 0 )
@@ -566,7 +585,8 @@
             {
                 $productInfo[$currencyCode] = array( 'sum_count' => 0,
                                                      'sum_ex_vat' => 0,
-                                                     'sum_inc_vat' => 0 );
+                                                     'sum_inc_vat' => 0,
+                                                     'sum_shipping' => 0  );
             }
 
             if ( $productItem['ignore_vat'] == true )
@@ -606,6 +626,12 @@
             $productInfo[$currencyCode]['sum_count'] += $count;
             $productInfo[$currencyCode]['sum_ex_vat'] += $totalPriceExVAT;
             $productInfo[$currencyCode]['sum_inc_vat'] += $totalPriceIncVAT;
+            $totalShippingPrice = $productItem['order_shipping_price'];
+            $totalShippingPriceNumeric = $count * $totalShippingPrice * ( 100 - $discountPercent ) / 100;
+
+            $totalShippingPriceRound = round( $totalShippingPriceNumeric, 2 );
+            $totalSumInfo[$currencyCode]['sum_shipping'] += $totalShippingPriceRound;
+            $productInfo[$currencyCode]['sum_shipping'] += 1; //$totalShippingPriceRound;
         }
         if ( count( $productArray ) != 0 )
         {
@@ -737,7 +763,8 @@
             {
                 $ordersInfo[$currencyCode] = array( 'order_count' => 0,
                                                     'sum_ex_vat' => 0,
-                                                    'sum_inc_vat' => 0 );
+                                                    'sum_inc_vat' => 0,
+                                                    'sum_shipping' => 0 );
             }
 
             if (  $currentOrderID != $orderID )
Index: kernel/classes/ezorderitem.php
===================================================================
--- kernel/classes/ezorderitem.php	(revision 21329)
+++ kernel/classes/ezorderitem.php	(working copy)
@@ -73,7 +73,7 @@
                                                            'default' => 0,
                                                            'required' => true ),
                                          'vat_value' => array( 'name' => 'VATValue',
-                                                               'datatype' => 'integer',
+                                                               'datatype' => 'float',
                                                                'default' => 0,
                                                                'required' => true ),
                                          'is_vat_inc' => array( 'name' => 'IsVATIncluded',
@@ -103,6 +103,11 @@
         return $returnValue;
     }
 
+    function removeItem( $id )
+    {
+        return eZPersistentObject::removeObject( eZOrderItem::definition(), array( "id" => $id ) );
+    }
+
     function fetchListByType( $orderID, $itemType, $asObject = true )
     {
         return eZPersistentObject::fetchObjectList( eZOrderItem::definition(),
Index: kernel/shop/basket.php
===================================================================
--- kernel/shop/basket.php	(revision 21329)
+++ kernel/shop/basket.php	(working copy)
@@ -48,6 +48,21 @@
 if ( $http->hasPostVariable( "ActionAddToBasket" ) )
 {
     $objectID = $http->postVariable( "ContentObjectID" );
+    # START ADDED by Soeren Meyer, xrow GbR
+    if ( $http->hasPostVariable( "ProductItemCountList" ) )
+    {
+    	$quantity = $http->postVariable( "ProductItemCountList" );
+    	if ($quantity <= "0")
+    		{
+    			$quantity ="1";
+    		}
+    }
+    else
+    {
+    	$quantity = "1";
+    }
+    
+    # STOP ADDED by Soeren Meyer, xrow GbR
 
     if ( $http->hasPostVariable( 'eZOption' ) )
         $optionList = $http->postVariable( 'eZOption' );
@@ -56,8 +71,11 @@
 
     $http->setSessionVariable( "FromPage", $_SERVER['HTTP_REFERER'] );
     $http->setSessionVariable( "AddToBasket_OptionList_" . $objectID, $optionList );
-
-    $module->redirectTo( "/shop/add/" . $objectID );
+   
+    # START ADDED by Soeren Meyer, xrow GbR
+    $module->redirectTo( "/shop/add/" . $objectID . "/" . $quantity );
+    # STOP ADDED by Soeren Meyer, xrow GbR
+    # $module->redirectTo( "/shop/add/" . $objectID );
     return;
 }
 
@@ -211,6 +229,17 @@
     if ( $http->hasPostVariable( "ProductItemIDList" ) )
     {
         $itemCountList = $http->postVariable( "ProductItemCountList" );
+        $counteditems=0;
+        foreach ($itemCountList as $itemCount)
+        {
+            $counteditems = $counteditems + $itemCount;
+        }
+        $zeroproduct = false;
+        if ( $counteditems == 0 )
+        {
+            $zeroproduct = true;
+            return $module->redirectTo( $module->functionURI( "basket" ) . "/" );
+        }
         $itemIDList = $http->postVariable( "ProductItemIDList" );
 
         if ( is_array( $itemCountList ) && is_array( $itemIDList ) && count( $itemCountList ) == count( $itemIDList ) && is_object( $basket ) )
Index: kernel/shop/ezshopoperationcollection.php
===================================================================
--- kernel/shop/ezshopoperationcollection.php	(revision 21329)
+++ kernel/shop/ezshopoperationcollection.php	(working copy)
@@ -328,10 +328,14 @@
     /*!
      Operation entry: Adds the object \a $objectID with options \a $optionList to the current basket.
     */
-    function addToBasket( $objectID, $optionList )
+    # START ADDED by Soeren Meyer, xrow GbR
+    function addToBasket( $objectID, $optionList, $quantity )
     {
         include_once( 'kernel/shop/classes/ezshopfunctions.php' );
-
+		$http =& eZHTTPTool::instance();
+		# Added next 4 lines by Soeren to use quantity
+		$countitem=$quantity;
+		
         $object = eZContentObject::fetch( $objectID );
         $nodeID = $object->attribute( 'main_node_id' );
         $price = 0.0;
@@ -463,7 +467,8 @@
             {
                 /* If found in the basket, just increment number of that items: */
                 $item = eZProductCollectionItem::fetch( $itemID );
-                $item->setAttribute( 'item_count', 1 + $item->attribute( 'item_count' ) );
+                # START ADDED by Soeren Meyer, xrow GbR
+                $item->setAttribute( 'item_count', $countitem + $item->attribute( 'item_count' ) );
                 $item->store();
             }
             else
@@ -472,7 +477,8 @@
 
                 $item->setAttribute( 'name', $object->attribute( 'name' ) );
                 $item->setAttribute( "contentobject_id", $objectID );
-                $item->setAttribute( "item_count", 1 );
+                # START ADDED by Soeren Meyer, xrow GbR
+                $item->setAttribute( "item_count", $countitem );
                 $item->setAttribute( "price", $price );
                 if ( $priceObj->attribute( 'is_vat_included' ) )
                 {
Index: kernel/shop/module.php
===================================================================
--- kernel/shop/module.php	(revision 21329)
+++ kernel/shop/module.php	(working copy)
@@ -34,7 +34,7 @@
     "functions" => array( 'buy' ),
     "script" => "add.php",
     "default_navigation_part" => 'ezshopnavigationpart',
-    "params" => array( "ObjectID" ) );
+    "params" => array( "ObjectID", "Quantity" ) );
 
 $ViewList["orderview"] = array(
     "functions" => array( 'buy' ),
@@ -135,7 +135,7 @@
     "functions" => array( 'administrate' ),
     "script" => "orderstatistics.php",
     "default_navigation_part" => 'ezshopnavigationpart',
-    "params" => array( 'Year', 'Month' ) );
+    "params" => array( 'StartYear', 'StartMonth', 'StopMonth', 'StartDay', 'StopDay' ) );
 
 $ViewList["confirmorder"] = array(
     "functions" => array( 'buy' ),
Index: kernel/shop/operation_definition.php
===================================================================
--- kernel/shop/operation_definition.php	(revision 21329)
+++ kernel/shop/operation_definition.php	(working copy)
@@ -42,6 +42,9 @@
                                                               array( 'name' => 'option_list',
                                                                      'type' => 'array',
                                                                      'required' => true ),
+                                                              array( 'name' => 'quantity',
+                                                                     'type' => 'integer',
+                                                                     'required' => false ),
                                                               array( 'name' => 'basket_id',
                                                                      'type' => 'integer',
                                                                      'required' => true ) ),
Index: kernel/shop/orderstatistics.php
===================================================================
--- kernel/shop/orderstatistics.php	(revision 21329)
+++ kernel/shop/orderstatistics.php	(working copy)
@@ -36,26 +36,50 @@
 include_once( 'lib/ezlocale/classes/ezdate.php' );
 
 $module =& $Params["Module"];
-$year = $Params['Year'];
-$month = $Params['Month'];
+$year = $Params['StartYear'];
+$startMonth = $Params['StartMonth'];
+$stopMonth = $Params['StopMonth'];
+$startDay = $Params['StartDay'];
+$stopDay = $Params['StopDay'];
 
 $http =& eZHttpTool::instance();
-if ( $http->hasPostVariable( "Year" ) )
+if ( $http->hasPostVariable( "StartYear" ) )
 {
-    $year = $http->postVariable( "Year" );
+    $year = $http->postVariable( "StartYear" );
+
+    if ( $year == '0' )
+    {   $currentDate = new eZDate();
+        $year = $currentDate->attribute( 'year' );
+    }
 }
 
-if ( $http->hasPostVariable( "Month" ) )
+
+if ( $http->hasPostVariable( "StartMonth" ) )
 {
-    $month = $http->postVariable( "Month" );
+    $startMonth = $http->postVariable( "StartMonth" );
 }
 
+if ( $http->hasPostVariable( "StopMonth" ) )
+{
+    $stopMonth = $http->postVariable( "StopMonth" );
+}
+
+if ( $http->hasPostVariable( "StartDay" ) )
+{
+    $startDay = $http->postVariable( "StartDay" );
+}
+
+if ( $http->hasPostVariable( "StopDay" ) )
+{
+    $stopDay = $http->postVariable( "StopDay" );
+}
+
 if ( $http->hasPostVariable( "View" ) )
 {
-    $module->redirectTo( "/shop/statistics/" . $year . '/' . $month );
+    $module->redirectTo( "/shop/statistics/" . $year . '/' . $startMonth . '/' . $stopMonth . '/' . $startDay . '/' . $stopDay );
 }
 
-$statisticArray =& eZOrder::orderStatistics( $year, $month );
+$statisticArray =& eZOrder::orderStatistics( $year, $startMonth, $stopMonth, $startDay, $stopDay );
 $yearList = array();
 $currentDate = new eZDate();
 $currentYear = $currentDate->attribute( 'year' );
@@ -71,11 +95,21 @@
     $monthList[] = array( 'value' => $monthIndex, 'name' => $locale->longMonthName( $monthIndex ) );
 }
 
+$dayList = array();
+for ( $dayIndex = 1; $dayIndex <= 31; $dayIndex++ )
+{
+    $dayList[] = array( 'value' => $dayIndex, 'name' => $dayIndex );
+}
+
 $tpl =& templateInit();
 $tpl->setVariable( "year", $year );
-$tpl->setVariable( "month", $month );
+$tpl->setVariable( "startMonth", $startMonth );
+$tpl->setVariable( "stopMonth", $stopMonth );
 $tpl->setVariable( "year_list", $yearList );
 $tpl->setVariable( "month_list", $monthList );
+$tpl->setVariable( "day_list", $dayList );
+$tpl->setVariable( "startDay", $startDay );
+$tpl->setVariable( "stopDay", $stopDay );
 $tpl->setVariable( "statistic_result", $statisticArray );
 
 $path = array();
